modules/core/src/v2/coins/xlm.ts

```diff
@@ -220,13 +223,20 @@ export class Xlm extends BaseCoin {
       throw new Error(`invalid address: ${address}`);
     }
 
+    if (Array.isArray(queryDetails.memoId) && queryDetails.memoId.length > 1) {
+      // valid addresses can only contain one memo id
+      throw new Error(`invalid address '${address}', can only contain one memoId`);
+    }
+
+    const memoId = queryDetails.memoId as string;
+
     try {
-      new BigNumber(queryDetails.memoId);
+      new BigNumber(memoId);
```

Comment:
Interesting, so we cast `queryDetails.memoId` to a string, and then pump it into BitNumber ... two thoughts:

1. Why cast it to a string in the first place? Seems like an int would be appropriate.
1. My understanding is that there are other memo types (MemoHash, MemoText, and one other ... i _think_). Are those supported or ignored?

@CapnMigraine can you shed any insight into the second point?

Reply:
We need to use it as a string later (isValidMemo expects a string), but we don't know if it's valid yet, so we try to create a big number from it. If that fails, we know the address is not valid. We need to cast to string because before that the type of `queryDetails.memoId` is `string | string[]` since a particular key can appear multiple times in a query string. This case was not handled explicitly before this change.

Reply:
We don't use numbers for memo ids because they take values up to 2^64 - 1, while JS numbers could lose precision, since they are capped at 2^53 - 1